cli/metrics/metrics.go

```diff
@@ -24,23 +24,32 @@ import (
 	"github.com/docker/compose/v2/pkg/utils"
 )
 
-// Track sends the tracking analytics to Docker Desktop
-func Track(context string, args []string, status string) {
+func (c *client) Track(context string, args []string, status string) {
 	if isInvokedAsCliBackend() {
 		return
 	}
 	command := GetCommand(args)
 	if command != "" {
-		c := NewClient()
 		c.Send(Command{
 			Command: command,
 			Context: context,
-			Source:  metadata.Get(CLISource, args),
+			Source:  c.getMetadata(CLISource, args),
 			Status:  status,
 		})
 	}
 }
 
+func (c *client) getMetadata(cliSource string, args []string) string {
+	if len(args) == 0 {
+		return cliSource
+	}
+	switch args[0] {
+	case "build", "buildx":
+		cliSource = metadata.BuildMetadata(cliSource, c.cliVersion(), args[0], args[1:])
```

Comment:
Calling this here will make build commands start slower (while we execute the cli to get the version), are we ok with this?

Reply:
Yes that's my concern too :disappointed:. Not sure how we can reliably returns the docker cli version without invocation.